tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -543,7 +550,7 @@ void TTreeProcessorMT::Process(std::function<void(TTreeReader &)> func)
       // Evaluate clusters (with local entry numbers) and number of entries for this file
       const auto &treeNames = std::vector<std::string>({fTreeNames[fileIdx]});
       const auto &fileNames = std::vector<std::string>({fFileNames[fileIdx]});
-      const auto clustersAndEntries = MakeClusters(treeNames, fileNames, maxTasksPerFile);
+      const auto clustersAndEntries = MakeClusters(treeNames, fileNames, maxTasksPerFile, fStartEntry, fEndEntry);
```

Comment:
How does this work here? `fStartEntry` and `fEndEntry` refer to global chain, but this lambda processes local files right?

Reply:
well, the boolean `shouldRetrieveAllClusters` now also checks `fStartEntry || fEndEntry != std::numeric_limits<Long64_t>::max()`, that is: it is checking if the user provided any range. So if the user provided some range, it will globally call `MakeClusters`.

This call here can happen only if no range is specified.

When there is no range specifed, `max(fStart = 0, currentStart)` will always give currentStart and similarly `min(fEnd=max, currentLast)` will always give the currentLast.

Reply:
This is now described inside of `MakeClusters`

Reply:
I agree with Vincenzo that it's confusing for the reader to pass `fStartEntry` and `fEndEntry` here, maybe we can have `MakeClusters` take an _optional_ `pair<Long64_t, Long64_t>` so we can just leave it out if we know we don't need to clamp the entry ranges to a certain start/end?

Reply:
Giving it a second look, "cleanest" within TTreeProcessorMT seems to me to ask for:
```cpp
struct EntryCluster {
   Long64_t start;
   Long64_t end;
};
```